core/cog-launcher.c

```diff
@@ -142,13 +162,23 @@ cog_launcher_activate (GApplication *application)
 #endif
 }
 
+static void
+cog_launcher_finalize (GObject *object)
+{
+    CogLauncher *launcher = COG_LAUNCHER (object);
+
+    g_free (launcher->permissions);
```

Comment:
Using a boolean or enum value (as suggested above) would make the `g_free` call unnecessary. Well, the `cog_launcher_finalize` function would not be needed, either.